monkestation/code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -19,3 +19,49 @@
 	gold_core_spawnable = NO_SPAWN
 	unique_pet = TRUE
 	speak = list("barks!", "woofs!", "borks!", "yips!", "burks!")
+
+/mob/living/simple_animal/pet/dog/corgi/borgi
+	name = "E-N"
+	real_name = "E-N"
+	desc = "It's a borgi."
+	icon = 'monkestation/icons/mob/pets.dmi'
+	icon_state = "borgi"
+	icon_living = "borgi"
+	icon_dead = "borgi_dead"
+	var/emagged = FALSE
+	minbodytemp = 0
+	loot = list(/obj/effect/decal/cleanable/robot_debris)
+	del_on_death = FALSE
+	deathmessage = "ceases to function"
+	animal_species = /mob/living/simple_animal/pet/dog/corgi/borgi
+	nofur = TRUE
+	var/blood_color = "#000000"
+	var/next_beep
+	var/chan
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/emag_act(mob/user)
+	if(!emagged)
+		emagged = TRUE
+		visible_message("<span class='warning'>[user] swipes a card through [src].</span>")
+		visible_message("<span class='notice'>You overload [src]s internal reactor.</span>")
+		visible_message("<span class='reallybig warning'>[src] makes an odd whining noise.</span>")
+		addtimer(CALLBACK(src, .proc/explode), 10 SECONDS, TIMER_STOPPABLE)
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/proc/explode()
+	explosion(get_turf(src), 0, 1, 4, 7)
+	death(src)
```

Comment:
When the dog explodes, the body remains.
Which means if you revive them later, they will beep forever.
My suggestion would be to set their emagged state to FALSE directly before the explosion.

Reply:
alright, also the reason why the body doesn't disappear anymore is cause i couldn't figure out a way for the runtime to stop whenever E-N's body was deleted and stuff, so i disabled E-N's body from being deleted and stuff, but yeah i'll try and figure out a way to stop the beeping from happening by setting their emagged state to false once after they explode so yeah